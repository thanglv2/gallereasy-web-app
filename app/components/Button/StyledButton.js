
import styled from 'styled-components';

const StyledButton = styled.button`
  color: #fff;
  background-color: ${props => props.theme.dodgerblue};
  border-color: ${props => props.theme.dodgerblue};
  text-shadow: 0 -1px 0 rgba(0,0,0,0.12);
  box-shadow: 0 2px 0 rgba(0,0,0,0.045);
  height: 40px;
  padding: 0 15px;
  font-size: 16px;
  border-radius: 4px;
  outline: 0;
  margin: auto;
  display: block;
  &:hover {
    cursor: pointer;
  }
`;

export default StyledButton;