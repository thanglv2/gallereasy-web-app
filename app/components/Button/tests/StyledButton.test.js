import React from 'react';
import renderer from 'react-test-renderer';

import StyledButton from '../StyledButton';

describe('<StyledButton />', () => {
  const onClick = jest.fn();

  it('render without error', () => {
    const theme = {
      dodgerblue: 'dodgerblue'
    }
    const tree = renderer.create(<StyledButton theme={theme} onClick={onClick} />).toJSON()
    expect(tree).toMatchSnapshot()
  });
});
