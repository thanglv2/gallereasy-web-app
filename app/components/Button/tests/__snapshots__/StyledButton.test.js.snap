// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledButton /> render without error 1`] = `
<button
  className="sc-bdVaJa jVOQiQ"
  onClick={[MockFunction]}
/>
`;
